libs/aws/langchain_aws/cache.py

```diff
@@ -0,0 +1,275 @@
+"""
+
+**Cache** provides an optional caching layer for LLMs.
+
+Cache is useful for two reasons:
+
+- It can save you money by reducing the number of API calls you make to the LLM
+  provider if you're often requesting the same completion multiple times.
+- It can speed up your application by reducing the number of API calls you make
+  to the LLM provider.
+
+Cache directly competes with Memory. See documentation for Pros and Cons.
+
+**Class hierarchy:**
+
+.. code-block::
+
+    BaseCache --> <name>Cache  # Examples: I
+"""
+
+from __future__ import annotations
```

Comment:
Would this code function without this statement?

Reply:
yes we are not implementing standard cache, only semantic caching.